v2/pkg/subscraping/utils.go

```diff
@@ -19,12 +25,34 @@ func NewSubdomainExtractor(domain string) (*regexp.Regexp, error) {
 	return extractor, nil
 }
 
-// Exists check if a key exist in a slice
-func Exists(values []string, key string) bool {
-	for _, v := range values {
-		if v == key {
-			return true
+func PickRandom[T any](v []T, sourceName string) T {
+	var result T
+	length := len(v)
+	if length == 0 {
+		gologger.Debug().Msgf("Cannot use the '%s' source because there was no API key/secret defined for it.", sourceName)
+		return result
+	}
+	return v[rand.Intn(length)]
```

Comment:
we should seed the random number generator first or the items picked won't be random by `rand.Intn`

Reply:
I haven't touched this part directly. The previous code used `sliceutil.PickRandom` which does exactly this. I've just extended it and made it generic.

```go
func PickRandom(v []string) string {
	return v[rand.Intn(len(v))]
}
```

Reply:
sliceutil does call rand.Seed internally automatically iirc

Reply:
> `sliceutil` does call `rand.Seed` internally automatically iirc

Actually it doesn't, but I'll update my code, because it's a good point.